src/common/collections.rs

```diff
@@ -63,21 +63,49 @@ pub async fn do_list_aliases(
 }
 
 pub async fn do_list_snapshots(
-    toc: &TableOfContent,
+    dispatcher: &Dispatcher,
+    collection_name: &str,
+    wait: bool,
+) -> Result<Vec<SnapshotDescription>, StorageError> {
+    let dispatcher = dispatcher.clone();
+    let collection_name = collection_name.to_string();
+    let task = tokio::spawn(async move { _do_list_snapshots(&dispatcher, &collection_name).await });
+
+    if wait {
+        task.await?
+    } else {
+        Ok(vec![])
+    }
+}
+
+async fn _do_list_snapshots(
+    dispatcher: &Dispatcher,
     collection_name: &str,
 ) -> Result<Vec<SnapshotDescription>, StorageError> {
-    Ok(toc
+    Ok(dispatcher
         .get_collection(collection_name)
         .await?
         .list_snapshots()
         .await?)
 }
 
-pub async fn do_create_snapshot(
-    toc: &TableOfContent,
+pub async fn do_create_snapshot<'a>(
+    dispatcher: &Dispatcher,
     collection_name: &str,
+    wait: bool,
 ) -> Result<SnapshotDescription, StorageError> {
-    toc.create_snapshot(collection_name).await
+    let collection = collection_name.to_string();
+    let dispatcher = dispatcher.clone();
+    let snapshot = tokio::spawn(async move { dispatcher.create_snapshot(&collection).await });
+    if wait {
+        Ok(snapshot.await??)
+    } else {
+        Ok(SnapshotDescription {
+            name: "".to_string(),
```

Comment:
I am unsure about using the snapshot description, empty or with message, to convey the semantic of the operation being in progress.

Ideally we should return a HTTP 202 I believe https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/202

WDYT?

Reply:
Yes, return 202 is a better solution. But I am not sure how to return StatusCode here.